src/new_game.c

```diff
@@ -0,0 +1,163 @@
+}
+
+static void SetDefaultOptions(void)
+{
+    gSaveBlock2Ptr->optionsTextSpeed = OPTIONS_TEXT_SPEED_MID;
+    gSaveBlock2Ptr->optionsWindowFrameType = 0;
+    gSaveBlock2Ptr->optionsSound = OPTIONS_SOUND_MONO;
+    gSaveBlock2Ptr->optionsBattleStyle = OPTIONS_BATTLE_STYLE_SHIFT;
+    gSaveBlock2Ptr->optionsBattleSceneOff = FALSE;
+    gSaveBlock2Ptr->regionMapZoom = FALSE;
+    gSaveBlock2Ptr->optionsButtonMode = OPTIONS_BUTTON_MODE_NORMAL;
+}
+
+static void ClearPokedexFlags(void)
+{
+    memset(&gSaveBlock2Ptr->pokedex.owned, 0, sizeof(gSaveBlock2Ptr->pokedex.owned));
+    memset(&gSaveBlock2Ptr->pokedex.seen, 0, sizeof(gSaveBlock2Ptr->pokedex.seen));
+}
+
+static void sub_80549D4(void)
+{
+    CpuFill32(0, &gSaveBlock2Ptr->field_B0, (u32) &gSaveBlock2Ptr->mapView - (u32) &gSaveBlock2Ptr->field_B0);
+}
+
+static void WarpToPlayersRoom(void)
+{
+    SetWarpDestination(MAP_GROUP(PALLET_TOWN_PLAYERS_HOUSE_2F), MAP_NUM(PALLET_TOWN_PLAYERS_HOUSE_2F), -1, 6, 6);
+    WarpIntoMap();
+}
+
+void Sav2_ClearSetDefault(void)
+{
+    ClearSav2();
+    SetDefaultOptions();
+}
+
+void ResetMenuAndMonGlobals(void)
+{
+    gDifferentSaveFile = 0;
+    ZeroPlayerPartyMons();
+    ZeroEnemyPartyMons();
+    sub_81089BC();
+    ResetTMCaseCursorPos();
+    BerryPouch_CursorResetToTop();
+    sub_811089C();
+    sub_8083214(Random());
+    sub_806E6FC();
+}
+
+void NewGameInitData(void)
+{
+    u8 rivalName[PLAYER_NAME_LENGTH];
+
+    StringCopy(rivalName, gSaveBlock1Ptr->rivalName);
+    gDifferentSaveFile = 1;
+    gSaveBlock2Ptr->encryptionKey = 0;
+    ZeroPlayerPartyMons();
+    ZeroEnemyPartyMons();
+    sub_80549D4();
+    ClearSav1();
+    ClearMailData();
+    gSaveBlock2Ptr->specialSaveWarpFlags = 0;
+    gSaveBlock2Ptr->field_A8 = 0;
+    gSaveBlock2Ptr->field_AC = 1;
+    gSaveBlock2Ptr->field_AD = 0;
+    InitPlayerTrainerId();
+    PlayTimeCounter_Reset();
+    ClearPokedexFlags();
+    InitEventData();
+    ResetFameChecker();
+    SetMoney(&gSaveBlock1Ptr->money, 3000);
+    ResetGameStats();
+    ClearPlayerLinkBattleRecords();
+    sub_80A0904();
+    sub_80A0958();
+    sub_806E190();
+    gPlayerPartyCount = 0;
+    ZeroPlayerPartyMons();
+    ResetPokemonStorageSystem();
+    ClearRoamerData();
+    gSaveBlock1Ptr->registeredItem = 0;
+    ClearBag();
+    NewGameInitPCItems();
+    sub_809C794();
+    InitEasyChatPhrases();
+    sub_8113044();
+    copy_strings_to_sav1();
+    ResetMiniGamesResults();
+    sub_8143D24();
+    sub_815D838();
+    WarpToPlayersRoom();
+    ScriptContext2_RunNewScript(EventScript_ResetAllMapFlags);
+    StringCopy(gSaveBlock1Ptr->rivalName, rivalName);
+    ResetTrainerTowerResults();
+}
+
+static void ResetMiniGamesResults(void)
+{
+    CpuFill16(0, &gSaveBlock2Ptr->berryCrush, sizeof(struct BerryCrush));
+    SetBerryPowder(&gSaveBlock2Ptr->berryCrush.berryPowderAmount, 0);
+    ResetPokeJumpResults();
+    CpuFill16(0, &gSaveBlock2Ptr->berryPick, sizeof(struct BerryPickingResults));
+}
+ 
```

Comment:
No newline at end of file

Reply:
I thought every file was supposed to have a new line?

Reply:
Every text file should end with newline. i.e. Line 163 should be empty.